issuetracker/internal/youtrack/task.go

```diff
@@ -16,7 +17,16 @@ type Task struct {
 
 // GetStatus of youtrack task, based on underlying structure
 func (t *Task) GetStatus() taskstatus.TaskStatus {
-	switch t.CustomFields[TaskStateIndex][Value].(map[string]interface{})[IsResolved].(bool) {
+	isResolved := false
+	// Loop through all fields to find status field
+	for _, node := range t.CustomFields {
+		if node[Type].(string) == StateType && node[Value] != nil {
+			isResolved = node[Value].(map[string]interface{})[IsResolved].(bool)
+			break
+		}
+	}
```

Comment:
In case the target state type & value is not found, we should return an error rather than silently returning that the issue is closed.

Could you modify the return of the function, associated interface & rest of implementations to return task status & error & then handle it accordingly in the caller?

The error can be:
```
couldn't fetch issue status from youtrack. This is probably on us, please file a bug report here: [link to opening an issue]
```

Reply:
This comment is on an old change which has already been addressed

Reply:
I started the review yesterday and forgot to submit it. Then I noticed today that I didn't submit it in. And I submit it by mistake. Sorry for the inconvenience.